tests/include/test/drivers/pake.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Test driver for PAKE driver entry points.
+ */
+/*  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef PSA_CRYPTO_TEST_DRIVERS_PAKE_H
+#define PSA_CRYPTO_TEST_DRIVERS_PAKE_H
+
+#include "mbedtls/build_info.h"
+
+#if defined(PSA_CRYPTO_DRIVER_TEST)
+#include <psa/crypto_driver_common.h>
+
+typedef struct {
+    /* If not PSA_SUCCESS, return this error code instead of processing the
+     * function call. */
+    psa_status_t forced_status;
+    /* PAKE driver setup is executed on the first call to
+       pake_output/pake_input (added to distinguish forced statuses). */
+    psa_status_t forced_setup_status;
+    /* Count the amount of times PAKE driver functions are called. */
+    unsigned long hits;
```

Comment:
I did it based on other test drivers and most of them have single hit counter, but all in one hit counter and counter per driver entry point sounds good.

Reply:
Yes I know some test drivers have only one global hit counter but I think that for the last ones we moved to one counter per entry point. That's a good trend I think thus the idea here is to keep promoting this new approach.

Reply:
Maybe structure for hits would be a good option, so we can reset counters by single assignment? Also it looks better I think.

```
struct{
   unsigned int total;
   unsigned int setup;
   unsigned int input;
...
} hits;
```

Reply:
I agree it looks good like that.

Reply:
In `pake_operations` checked both `total` counter and entry point counter to confirm that all functions are called as expected.

In `ecjpake_rounds` checked only `total` counter as this would require a lot of changes and since we have confirmed that entry points are called as expected in `pake_operations` relaying on `total` counter here should be enough.